src/Type/Php/ThrowableReturnTypeExtension.php

```diff
@@ -50,14 +51,16 @@ public function getTypeFromMethodCall(MethodReflection $methodReflection, Method
 					continue 2;
 				}
 			}
+
 			if ($pdoException->isSuperTypeOf($classType)->yes()) {
-				$types[] = new StringType();
+				$types[] = new UnionType([new IntegerType(), new StringType()]);
```

Comment:
I'd rather be if this was a benevolent union type.

Reply:
Sure, I made the change